source/YTEPs/YTEP-0036.rst

```diff
@@ -0,0 +1,248 @@
+
+  $ pytest -s -v -k "TestClass"
+
+and pytest will collect every test and then ignore all of the tests not contained in the ``TestClass`` class. To run only a specific method within a given class, one would do:
+
+.. code-block:: bash
+
+  $ pytest -s -v -k "TestClass and test_method"
+
+See `this link <https://docs.pytest.org/en/latest/usage.html#specifying-tests-selecting-tests>`_ for more on pytest's selection capabilities and options.
+
+Answer Tests
+""""""""""""
+
+To run the answer tests previously, one had to first tell yt where the test data used in the answer tests was located via
+
+.. code-block:: bash
+
+  $ yt config set yt test_data_dir /path/to/yt-data
+
+This remains true here, as well.
+
+Previously, to run the answer tests for a specific frontend, one would do:
+
+.. code-block:: bash
+
+  nosetests --with-answer-testing --local --local-dir $HOME/Documents/test --answer-store --answer-name=local-tipsy yt.frontends.tipsy
+
+This command would tell nose to select the answer tests, use the local test data, save the generated answers in the specified ``--local-dir`` with the name ``local-tipsy``, and then run the tipsy frontend answer tests.
+
+Now, with pytest, this same action is performed with
+
+.. code-block:: bash
+
+  $ pytest --with-answer-testing --answer-store -k "TestTipsy"
+
+Since the answers are now packaged with the code, they are saved to ``yt_repo/tests/answers`` by default.
+
+Previously, the answer files associated with each test were contained in ``yt_repo/tests/tests.yaml``. This remains the case, but the structure of that file has been changed. Previously, ``tests.yaml`` had the format
+
+.. code-block:: yaml
+
+  answer_file_name_xyz:
+    - /path/to/file1/that/uses/these/answers.py
+    - /path/to/file2/that/uses/these/answers.py
+    ...
+
+Now the format is:
+
+.. code-block:: yaml
+
+  TestClassName1:
+    answer_file_name1_xyz.yaml
+  TestClassName2:
+    answer_file_name2_xyz.yaml
+  ...
+
+The ``xyz`` identifier is used to identify certain answer changesets. Comparing against a different answer changeset requires only changing that identifier in the ``tests.yaml`` file.
+
+There are several answer tests (e.g., ``TestArt.test_d9p``) that use particularly large data files. Previously, to run these tests, they had to be specifically selected via the ``--answer-big-data`` command line option. This remains the case:
+
+.. code-block:: bash
+
+  $ pytest --with-answer-testing --answer-store --answer-big-data -k "TestArt and test_d9p"
+
+Writing New Tests
+^^^^^^^^^^^^^^^^^
+
+If one is writing tests for a new frontend, one should create a ``tests`` directory inside of the new frontend directory and, within the ``tests`` directory, create a file called ``test_outputs.py``. Within this file, one should define a new class that inherits from ``AnswerTest`` and is marked by pytest as being an answer test. All of the frontend tests should now be methods of this class. For example:
+
+.. code-block:: python
```

Comment:
note that this Python example code doesn't follow yt's code style (mostly black)
Fortunately, you can use this tool to get there in a second https://pypi.org/project/blacken-docs/

Reply:
@jcoughlin11 the test method still has a clunky indentation. Please resolve this thread when you're done updating it :)